ts/nni_manager/training_service/reusable/environments/kubernetes/frameworkcontrollerEnvironmentService.ts

```diff
@@ -0,0 +1,316 @@
+                    labels: {
+                        app: this.NNI_KUBERNETES_TRIAL_LABEL,
+                        expId: this.experimentId,
+                        trialId: trialJobId
+                    }
+                },
+                spec: {
+                    executionType: 'Start',
+                    taskRoles: taskRoles
+                }
+            });
+        }
+    
+        private generateTaskRoleConfig(trialWorkingFolder: string, replicaImage: string, runScriptFile: string,
+            podResources: any, containerPort: number, privateRegistrySecretName: string | undefined): any {
+    
+            const volumeSpecMap: Map<string, object> = new Map<string, object>();
+            if (this.config.storage.storageType === 'azureStorage') {
+                volumeSpecMap.set('nniVolumes', [
+                    {
+                        name: 'nni-vol',
+                        azureFile: {
+                            secretName: `${this.azureStorageSecretName}`,
+                            shareName: `${this.azureStorageShare}`,
+                            readonly: false
+                        }
+                    }, {
+                        name: 'frameworkbarrier-volume',
+                        emptyDir: {}
+                    }]);
+            } else {
+                volumeSpecMap.set('nniVolumes', [
+                    {
+                        name: 'nni-vol',
+                        nfs: {
+                            server: `${this.config.storage.server}`,
+                            path: `${this.config.storage.path}`
+                        }
+                    }, {
+                        name: 'frameworkbarrier-volume',
+                        emptyDir: {}
+                    }]);
+            }
+    
+            const containers: any = [
+                {
+                    name: 'framework',
+                    image: replicaImage,
+                    command: ['sh', `${path.join(trialWorkingFolder, runScriptFile)}`],
+                    volumeMounts: [
+                        {
+                            name: 'nni-vol',
+                            mountPath: this.CONTAINER_MOUNT_PATH
+                        }, {
+                            name: 'frameworkbarrier-volume',
+                            mountPath: '/mnt/frameworkbarrier'
+                        }],
+                    resources: podResources,
+                    ports: [{
+                        containerPort: containerPort
+                    }]
+                }];
+    
+            const initContainers: any = [
+                {
+                    name: 'frameworkbarrier',
+                    image: 'frameworkcontroller/frameworkbarrier',
+                    volumeMounts: [
+                        {
+                            name: 'frameworkbarrier-volume',
+                            mountPath: '/mnt/frameworkbarrier'
+                        }]
+                }];
```

Comment:
These configs look redundant. Would be nice to save repetitive parts into variables and put them in separate place.